### PR TITLE
Deprecating `body` and replacing with `response_body`

### LINKED ISSUE #10: Follow http and deprecate body in favor of response_body
`body` in `http` is deprecated, and replaced by `response_body`. Please follow it in this module.

See hashicorp/terraform-provider-http#137

Thanks.

### PR DESCRIPTION
**Pull request by orgads**
Cherry picked from hashicorp/terraform-provider-http#137.

Fixes #10.

### CODE REVIEW

File: internal/provider/data_source.go [hunk 1]
```diff
@@ -79,11 +76,17 @@ func dataSource() *schema.Resource {
...
-				Type:     schema.TypeString,
-				Computed: true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
-				},
+				Description: "The raw body of the HTTP response. " +
+					"**NOTE**: This is deprecated, use `response_body` instead.",
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: "Use response_body instead",
```
Comment (salrashid123) [inline internal/provider/data_source.go:83]: lets keep 

```golang
				Elem: &schema.Schema{
					Type: schema.TypeString,
				},
```

in the `body` and `request_body`.    I do know in the upstream provider [here](https://github.com/hashicorp/terraform-provider-http/blob/main/internal/provider/data_source_http.go#L208-L209) the Elem isn't set but later on they retun the datasrource value as a type.  if its truly not necessary, we can clean up all references in another PR

**Review by salrashid123 — CHANGES_REQUESTED** [review]
thatnks for the PR.

Couple of changes to the test cases.   The terraform that a testcase exercises must have a corresponding line of code that checks for it.


you can run the tests using `make tests`

```
terraform-provider-http-full$ make test
==> Checking that code complies with gofmt requirements...
go test -i $(go list ./...) || exit 1
go: -i flag is deprecated
echo $(go list ./...) | \
	xargs -t -n4 go test  -timeout=30s -parallel=4
go test '-timeout=30s' '-parallel=4' github.com/salrashid123/terraform-provider-http-full github.com/salrashid123/terraform-provider-http-full/internal/provider
?   	github.com/salrashid123/terraform-provider-http-full	[no test files]
ok  	github.com/salrashid123/terraform-provider-http-full/internal/provider	6.118s

```

File: internal/provider/data_source.go (hunk 1 above)
Comment (orgads) [reply internal/provider/data_source.go:83]: Sure, done

File: internal/provider/data_source_test.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ data "http" "http_test" {
 }
 
 output "body" {
-  value = data.http.http_test.body
+  value = data.http.http_test.response_body
```
[the remark this answers is not in the document]
Comment (orgads) [reply internal/provider/data_source_test.go:32]: Done

File: internal/provider/data_source_test.go [hunk 3]
```diff
@@ -164,7 +171,7 @@ data "http" "http_test" {
 }
 
 output "body" {
```
[the remark this answers is not in the document]
Comment (orgads) [reply internal/provider/data_source_test.go:173]: Done

**Comment by orgads** [comment]
Thank you. The tests pass now.

**Comment by salrashid123** [comment]
thanks,  this and the PRs 12,13 included in

```hcl
terraform {
  required_providers {
    http-full = {
      source = "salrashid123/http-full"
            version = "1.3.1"
    }
  }
}
```

**Pull request merged**